src/compile/scale.ts

```diff
@@ -122,7 +123,7 @@ function parseBinColorLegendLabel(model: Model, fieldDef: FieldDef) {
     type: 'ordinal',
     domain: {
       data: model.dataTable(),
-      field: model.field(COLOR,  {prefn: 'rank_'}),
+      field: model.field(COLOR, (fieldDef.bin || fieldDef.timeUnit) ? {} : { prefn: 'rank_' }),
```

Comment:
@domoritz   Now I know when this got wrong.  

On this line, `fieldDef.bin` is always `true`.  (See how `parseBinColorLegendLabel` gets called.)  So I asked you to remove `fieldDef.bin`, but I guess you put in the wrong condition.


I think the right fix here is `model.field(COLOR, {})`